Task/devguide release (#1006)
LLNL/RAJA: docs/sphinx/dev_guide/contributing.rst

capabilities. This section describes basic processes to follow
for individuals outside of the core RAJA team to contribute new features or 
bugfixes to RAJA. It assumes you are familiar with 
`Git <https://git-scm.com/>`_, which we use for source code version control,
and `GitHub <https://github.com/>`_, which is where our project is hosted. 

This section describes development processes, such as:

  * Making a fork of the RAJA repository 
  * Creating a branch for development
  * Creating a pull request (PR)
  * Tests that your PR must pass before it can be merged into RAJA

============
Forking RAJA
============

If you are not a member of the LLNL organization on GitHub or not a member of 
the core RAJA team of developers, then you do not have permission to create 
a branch in the RAJA repository. This is due to the policy adopted by the LLNL
organization on GitHub in which the RAJA project resides. Fortunately, you may 
still contribute to RAJA by `forking the RAJA repo 
<https://github.com/LLNL/RAJA/fork>`_. Forking creates a copy of the RAJA 
repository that you own. You can push code changes on that copy to GitHub and 
create a pull request in the RAJA project.

> .. note:: A contributor who is not a member of the LLNL GitHub organization 
>           and a member of the core team of RAJA developers cannot create a

Reviewer: ```suggestion
.. note:: A contributor who is not a member of the LLNL GitHub organization 
          or who is not a member of the core team of RAJA developers cannot create a
```

Author: Got it.

          branch in the RAJA repo. However, anyone can create a fork of the 
          RAJA project and create a pull request in the RAJA project.

=========================
Developing RAJA Code
=========================

> New features, bugfixes, and other changes are developed on a **feature branch**

Reviewer: ```suggestion
New features, bugfixes, and other changes are developed on a **feature branch**.
```

the ``RAJA/docs`` directory.

After your new code is complete, you've tested it, and developed appropriate
documentation, you can push your branch to GitHub and create a PR in the RAJA
project. It will be reviewed by members of the RAJA team, who will provide 
comments, suggestions, etc. After it is approved and all CI checks pass, your 
contribution will be merged into the RAJA repository.

.. important:: When creating a branch that you intend to be merged into the 
               RAJA repo, please give it a succinct name that clearly describes 
               the contribution.  For example, **feature/<name-of-feature>** 
               for a new feature, **bugfix/<fixed-issue>** for a bugfix, etc.

--------------------
Developing a Bug Fix
--------------------

Contributing a bugfix follows the same process as described above. Be sure to
indicate in the name of your branch that it is for a bugfix; for example:

.. code-block:: bash

    $ git checkout -b bugfix/<fixed-issue>

We recommend that you add a test that reproduces the issue you have found

> and which demonstrates that issue is resolved. To verify that you have done

Reviewer: ```suggestion
and that demonstrates that the issue is resolved. To verify that you have done
```

Author: Got it.

and any other relevant details that will assist the RAJA team in reviewing your
contribution.

When a PR is created in RAJA, it will be run through our automated testing

> processes and be reviewed by RAJA team members. Providing the PR passes all 

Reviewer: ```suggestion
processes and be reviewed by RAJA team members. Provided that the PR passes all 
```

Author: Got it.

Due to LLNL security policies and RAJA project policies, only a PR created
by someone on the RAJA core development team will be run automatically

> through all RAJA CI tools. In particular, a PR made from branch on a forked 
> repository will not. Gitlab CI on internal LLNL platforms and Travis CI will 

Reviewer: (I thought this was worded oddly, maybe this would be better?)
```suggestion
through all RAJA CI tools. In particular, a PR made from a branch on a forked 
repository will not automatically trigger the RAJA CI. Gitlab CI on internal LLNL platforms and Travis CI will 
```

Author: Got it.

          If successful, this will create a branch in your local copy of the

>           RAJA repo labeled ``pr-from-fork/<PR #> and you will be on that

Reviewer: Is there a closing `` missing from here? It doesn't seem to mess up the file when I view it, but it looks like it's missing the ``...

Reviewer: Lol I don't know how to type them without git thinking I want to type code...

Author: Yup. Got it. Thanks.

          local branch in your checkout space. To verify this, you can run
          the following command after you run the script::

            $ git branch

          You will see the new branch in the listing of branches and the branch
          you are on will be starred.

          You can push the new branch to the RAJA repo on GitHub::

            $ git push origin <branch-name>

          and make a PR for the new branch. It is good practice to reference 
          the original PR in the description of the new PR to track the 
          original PR discussion and reviews.

          All CI checks will be triggered to run on the PR make in the

>           RAJA repo. When everything passes and the PR is approved, it may 

Reviewer: The first sentence has a typo, but I'm not sure how it's supposed to be structured

Author: Fixed.

          be merged. When it is merged, the original PR from the forked repo 

>           will be closed and marked as merged also unless it is referenced 

Reviewer: ```suggestion
          will be closed and marked as merged unless it is referenced 
```

Author: Got it.
